homeassistant/components/bluetooth/websocket_api.py

```diff
@@ -0,0 +1,163 @@
+        # Keep time_diff precise to 2 decimal places
+        # so the cached serialization can be reused,
+        # however we still want to calculate it each
+        # subscription in case the system clock is wrong
+        # and gets corrected.
+        self.time_diff = round(time.time() - time.monotonic(), 2)
+
+    @callback
+    def _async_unsubscribe(
+        self, cancel_callbacks: tuple[Callable[[], None], ...]
+    ) -> None:
+        """Unsubscribe the callback."""
+        for cancel_callback in cancel_callbacks:
+            cancel_callback()
+
+    @callback
+    def async_start(self) -> None:
+        """Start the subscription."""
+        connection = self.connection
+        cancel_adv_callback = async_register_callback(
+            self.hass,
+            self._async_on_advertisement,
+            self.match_dict,
+            BluetoothScanningMode.PASSIVE,
+        )
+        cancel_disappeared_callback = _get_manager(
+            self.hass
+        ).async_register_disappeared_callback(self._async_removed)
+        connection.subscriptions[self.ws_msg_id] = partial(
+            self._async_unsubscribe, (cancel_adv_callback, cancel_disappeared_callback)
+        )
+        self.pending = False
+        self.connection.send_message(
+            json_bytes(websocket_api.result_message(self.ws_msg_id))
+        )
+        self._async_added(self.pending_service_infos)
+        self.pending_service_infos.clear()
+
+    def _async_added(self, service_infos: Iterable[BluetoothServiceInfoBleak]) -> None:
+        self.connection.send_message(
+            json_bytes(
+                websocket_api.event_message(
+                    self.ws_msg_id,
+                    {
+                        "add": [
+                            serialize_service_info(service_info, self.time_diff)
+                            for service_info in service_infos
+                        ]
+                    },
+                )
+            )
+        )
+
+    def _async_removed(self, address: str) -> None:
+        self.connection.send_message(
+            json_bytes(
+                websocket_api.event_message(
+                    self.ws_msg_id,
+                    {
+                        "remove": [
+                            {
+                                "address": address,
+                            }
+                        ]
+                    },
+                )
+            )
+        )
+
+    @callback
+    def _async_on_advertisement(
+        self, service_info: BluetoothServiceInfoBleak, change: BluetoothChange
+    ) -> None:
+        """Handle the callback."""
+        if self.pending:
+            self.pending_service_infos.append(service_info)
+            return
+        self._async_added((service_info,))
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "bluetooth/subscribe_advertisements",
+    }
+)
+@websocket_api.async_response
+async def ws_subscribe_advertisements(
```

Comment:
add `require_admin` decorator

Reply:
Discussed on discord. This is all public information floating through the air so no need to be an admin. When we do active/passive settings for the adapter, that will need to be admin only